provider/ec2/environ.go

```diff
@@ -1918,24 +1918,32 @@ func rulesToIPPerms(rules firewall.IngressRules) []types.IpPermission {
 			ToPort:     aws.Int32(int32(r.PortRange.ToPort)),
 		}
 		if len(r.SourceCIDRs) == 0 {
-			ipPerms[i].IpRanges = []types.IpRange{{CidrIp: aws.String(defaultRouteCIDRBlock)}}
-			ipPerms[i].Ipv6Ranges = []types.Ipv6Range{{CidrIpv6: aws.String(defaultRouteIPv6CIDRBlock)}}
+			ipPerms[i].IpRanges = []types.IpRange{{CidrIp: aws.String(defaultRouteCIDRBlock), Description: ipRangeDescription(r, defaultRouteCIDRBlock)}}
+			ipPerms[i].Ipv6Ranges = []types.Ipv6Range{{CidrIpv6: aws.String(defaultRouteIPv6CIDRBlock), Description: ipRangeDescription(r, defaultRouteIPv6CIDRBlock)}}
 		} else {
 			for _, cidr := range r.SourceCIDRs.SortedValues() {
 				// CIDRs are pre-validated; if an invalid CIDR
 				// reaches this loop, it will be skipped.
 				addrType, _ := network.CIDRAddressType(cidr)
 				if addrType == network.IPv4Address {
-					ipPerms[i].IpRanges = append(ipPerms[i].IpRanges, types.IpRange{CidrIp: aws.String(cidr)})
+					ipPerms[i].IpRanges = append(ipPerms[i].IpRanges, types.IpRange{CidrIp: aws.String(cidr), Description: ipRangeDescription(r, cidr)})
 				} else if addrType == network.IPv6Address {
-					ipPerms[i].Ipv6Ranges = append(ipPerms[i].Ipv6Ranges, types.Ipv6Range{CidrIpv6: aws.String(cidr)})
+					ipPerms[i].Ipv6Ranges = append(ipPerms[i].Ipv6Ranges, types.Ipv6Range{CidrIpv6: aws.String(cidr), Description: ipRangeDescription(r, cidr)})
 				}
 			}
 		}
 	}
 	return ipPerms
 }
 
+func ipRangeDescription(rule firewall.IngressRule, cidr string) *string {
+	if cidr == "" || cidr == firewall.AllNetworksIPV4CIDR || cidr == firewall.AllNetworksIPV6CIDR {
+		return aws.String(fmt.Sprintf("juju ingress to %v", rule.PortRange))
+	}
+	return aws.String(fmt.Sprintf("juju ingress to %v from %v", rule.PortRange, cidr))
```

Comment:
`cidr` is a string so think we should use `%s` or `%q`. My vote would be for %s in this situation.

rule.PortRange also implements Stringer so we can use `%s`

Reply:
I can never remember the difference so I often just default to %v :sweat_smile: